pvlib/location.py

```diff
@@ -85,46 +85,79 @@ def __repr__(self):
             ('{}: {}'.format(attr, getattr(self, attr)) for attr in attrs)))
 
     @classmethod
-    def from_tmy(cls, tmy_metadata, tmy_data=None, **kwargs):
+    def from_tmy_or_epw(cls, weather_metadata, weather_data=None, **kwargs):
         """
         Create an object based on a metadata
-        dictionary from tmy2 or tmy3 data readers.
+        dictionary from tmy2, tmy3 or epw data readers.
 
         Parameters
         ----------
-        tmy_metadata : dict
-            Returned from tmy.readtmy2 or tmy.readtmy3
-        tmy_data : None or DataFrame, default None
-            Optionally attach the TMY data to this object.
+        weather_metadata : dict
+            Returned from tmy.readtmy2, tmy.readtmy3 or epw.read_epw
```

Comment:
this diff makes me realize the doc string will need to be updated to `read_tmy2` and `read_tmy3`